### PR TITLE
Fix GridIn::read_vtk().

### PR DESCRIPTION
**Pull request by bangerth**
The function reads a section of VTK files of the form

CELL_DATA 4
SCALARS MaterialID double 1
LOOKUP_TABLE default
1.000000
2.000000
3.000000
4.000000

but internally (despite the fact that the field is declared
as 'double' above) reads them as integers. This leads to
wrong results if they are given as floating point numbers,
as in the case above.

While there also add an assertion an clean things up
slightly.

### CODE REVIEW

**Comment by bangerth** [comment]
Fixes #3261.

File: source/grid/grid_in.cc [hunk 1]
```diff
@@ -331,7 +350,7 @@ void GridIn<dim, spacedim>::read_vtk(std::istream &in)
 
           for (unsigned int i = 0; i < no_cells; i++) //assigning IDs to cells.
             {
-              int id;
+              double id;
```
Comment (bangerth) [inline source/grid/grid_in.cc]: this line was the bug, along with the same lines below

**Comment by drwells** [comment]
To be clear: the bug was that `1.0000` was parsed as `0`, right?

**Comment by bangerth** [comment]
No, the bug was that the `1` was parsed as an integer (correctly as one), but then the whole `istream` machinery hung trying to find more integers but only finding a period :-) The stream wouldn't yield any more information, and every execution of `input >> id` simply left `id` in an uninitialized state. I found the issue because on my system, `id` was set to zero after the first one, whereas on other systems it was left at one. The output of the test was consequently different on my system than what we save in the `.output` file.

**Pull request merged**